pkg/console/tlsconfig/tlsconfig.go

```diff
@@ -0,0 +1,281 @@
+	}
+}
+
+type watch struct {
+	running atomic.Bool
+	lock    sync.RWMutex
+
+	tlsProfilePath string
+	certsPath      string
+	keyPath        string
+
+	tlsProfileError error
+	ciphers         []uint16
+	minTlsVersion   uint16
+
+	certificate *tls.Certificate
+	certError   error
+}
+
+func (w *watch) Run(done <-chan struct{}) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf("could not create fsnotify.Watcher: %w", err)
+	}
+	// watcher.Close() never returns an error
+	defer func() { _ = watcher.Close() }()
+
+	err = watcher.Add(w.tlsProfilePath)
+	if err != nil {
+		return fmt.Errorf("could not add watch: %w", err)
+	}
+
+	err = watcher.Add(w.certsPath)
+	if err != nil {
+		return fmt.Errorf("failed to watch Certificate: %w", err)
+	}
+
+	err = watcher.Add(w.keyPath)
+	if err != nil {
+		return fmt.Errorf("failed to watch Certificate key: %w", err)
+	}
+
+	return w.processEvents(watcher, done)
+}
+
+func (w *watch) IsRunning() bool {
+	return w.running.Load()
+}
+
+func (w *watch) Reload() {
+	w.reloadTlsProfile()
+	w.reloadCertificate()
+}
+
+func (w *watch) GetConfig() (*tls.Config, error) {
+	w.lock.RLock()
+	defer w.lock.RUnlock()
+
+	if w.tlsProfileError != nil {
+		return nil, w.tlsProfileError
+	}
+	if w.certError != nil {
+		return nil, w.certError
+	}
+
+	if w.certificate == nil {
+		return nil, fmt.Errorf("certificate not loaded")
+	}
+
+	return &tls.Config{
+		CipherSuites: w.ciphers,
+		MinVersion:   w.minTlsVersion,
+		Certificates: []tls.Certificate{*w.certificate},
+	}, nil
+}
+
+func (w *watch) processEvents(watcher *fsnotify.Watcher, done <-chan struct{}) error {
+	w.running.Store(true)
+	defer w.running.Store(false)
+
+	for {
+		select {
+		case <-done:
+			return nil
+
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return nil
+			}
+			w.handleEvent(event)
+
+		case err, ok := <-watcher.Errors:
+			if !ok {
+				return nil
+			}
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (w *watch) handleEvent(event fsnotify.Event) {
+	const modificationEvents = fsnotify.Create | fsnotify.Write | fsnotify.Remove
+	if event.Op&modificationEvents == 0 {
+		return
+	}
+
+	switch {
+	case strings.HasPrefix(w.tlsProfilePath, event.Name):
+		w.reloadTlsProfile()
+	case strings.HasPrefix(w.certsPath, event.Name):
+		w.reloadCertificate()
+	case strings.HasPrefix(w.keyPath, event.Name):
+		w.reloadCertificate()
+	}
+}
+
+func (w *watch) reloadTlsProfile() {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	w.tlsProfileError = nil
+
+	ciphers, minVersion, err := loadCipherSuitesAndMinVersion(w.tlsProfilePath)
+	if errors.Is(err, os.ErrNotExist) {
+		// Config file does not exist, using zero values for default
+		w.ciphers = nil
+		w.minTlsVersion = 0
+		return
+	}
+	if err != nil {
+		log.Log.Errorf("Failed to load TLS configuration: %s", err)
+		w.tlsProfileError = fmt.Errorf("failed to load TLS configuration: %w", err)
+		return
+	}
+
+	log.Log.Infof("Loaded TLS configuration.")
```

Comment:
maybe it would be good to add here a debug line with info what was set.

Reply:
+1